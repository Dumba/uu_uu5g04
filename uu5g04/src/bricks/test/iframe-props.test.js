/**
 * Copyright (C) 2021 Unicorn a.s.
 *
 * This program is free software: you can redistribute it and/or modify it under the terms of the GNU General Public
 * License as published by the Free Software Foundation, either version 3 of the License, or (at your option) any later
 * version.
 *
 * This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without even the implied
 * warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU General Public License at
 * <https://gnu.org/licenses/> for more details.
 *
 * You may obtain additional information at <https://unicorn.com> or contact Unicorn a.s. at address: V Kapslovne 2767/2,
 * Praha 3, Czech Republic or at the email: info@unicorn.com.
 */

import UU5 from "uu5g04";
import "uu5g04-bricks";

const { mount, shallow, wait } = UU5.Test.Tools;

const CONFIG = {
  mixins: [
    "UU5.Common.BaseMixin",
    "UU5.Common.ElementaryMixin",
    "UU5.Common.NestingLevelMixin",
    "UU5.Common.PureRenderMixin",
  ],
  props: {
    src: {
      values: ["https://plus4u.net/"],
    },
    resize: {
      values: [true, false],
    },
    height: {
      values: ["400", "40em"],
    },
    syncTimeout: {
      values: [3],
      // make snapshots without waiting for timeout (to see that iframe is not present initially)
      opt: { wait: false },
    },
    inline: {
      values: [true, false],
    },
    iframeAttrs: {
      values: [{ name: "reactName" }],
    },
  },
  requiredProps: {},
  opt: {
    shallowOpt: {
      disableLifecycleMethods: false,
    },
  },
};

describe(`UU5.Bricks.Iframe`, () => {
  UU5.Test.Tools.testProperties(UU5.Bricks.Iframe, CONFIG);
});

describe(`UU5.Bricks.Iframe docKit examples`, () => {
  it(`UU5.Bricks.Iframe example01`, () => {
    const wrapper = shallow(
      <UU5.Bricks.Container id={"uuID"}>
        <UU5.Bricks.Column id={"uuID2"} colWidth="s6">
          <UU5.Bricks.Iframe id={"uuID3"} />
        </UU5.Bricks.Column>
        <UU5.Bricks.Column id={"uuID4"} colWidth="s6">
          <UU5.Bricks.Iframe id={"uuID5"} src="https://plus4u.net" />
        </UU5.Bricks.Column>
      </UU5.Bricks.Container>
    );
    expect(wrapper).toMatchSnapshot();
  });
});
