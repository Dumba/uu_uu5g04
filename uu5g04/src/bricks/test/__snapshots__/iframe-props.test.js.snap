// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UU5.Bricks.Iframe className 1`] = `
<div
  className="uu5-bricks-iframe my-classname"
  id="uuID"
  style={Object {}}
  title={null}
/>
`;

exports[`UU5.Bricks.Iframe controlled 1`] = `
<div
  className="uu5-bricks-iframe"
  id="uuID"
  style={Object {}}
  title={null}
/>
`;

exports[`UU5.Bricks.Iframe controlled 2`] = `
<div
  className="uu5-bricks-iframe"
  id="uuID"
  style={Object {}}
  title={null}
/>
`;

exports[`UU5.Bricks.Iframe controlled 3`] = `
<div
  className="uu5-bricks-iframe"
  id="uuID"
  style={Object {}}
  title={null}
/>
`;

exports[`UU5.Bricks.Iframe controlled 4`] = `
<div
  className="uu5-bricks-iframe"
  id="uuID"
  style={Object {}}
  title={null}
/>
`;

exports[`UU5.Bricks.Iframe default props 1`] = `
<div
  className="uu5-bricks-iframe"
  id="uuID"
  style={Object {}}
  title={null}
/>
`;

exports[`UU5.Bricks.Iframe disabled 1`] = `
<div
  className="uu5-bricks-iframe uu5-common-disabled"
  id="uuID"
  style={Object {}}
  title={null}
>
  <span
    className="uu5-common-disabled-cover"
    key="disabledCover"
  />
</div>
`;

exports[`UU5.Bricks.Iframe disabled 2`] = `
<div
  className="uu5-bricks-iframe"
  id="uuID"
  style={Object {}}
  title={null}
/>
`;

exports[`UU5.Bricks.Iframe disabled 3`] = `
<div
  className="uu5-bricks-iframe"
  id="uuID"
  style={Object {}}
  title={null}
/>
`;

exports[`UU5.Bricks.Iframe disabled 4`] = `
<div
  className="uu5-bricks-iframe uu5-common-disabled"
  id="uuID"
  style={Object {}}
  title={null}
>
  <span
    className="uu5-common-disabled-cover"
    key="disabledCover"
  />
</div>
`;

exports[`UU5.Bricks.Iframe docKit examples UU5.Bricks.Iframe example01 1`] = `
<div
  className="uu5-bricks-container uu5-bricks-container-spacing"
  id="uuID"
  style={Object {}}
  title={null}
>
  <Column
    className={null}
    colWidth="s6"
    content={null}
    controlled={true}
    disabled={false}
    dynamic={false}
    footer={null}
    header={null}
    hidden={false}
    id="uuID2"
    ignoreInnerHTML={false}
    key="uuID2"
    level={null}
    mainAttrs={null}
    name={null}
    nestingLevel={null}
    noIndex={false}
    noSpacing={false}
    pureRender={false}
    ref_={null}
    selected={false}
    style={null}
    tooltip={null}
    underline={false}
    width={null}
  >
    <Iframe
      className={null}
      controlled={true}
      disabled={false}
      height="250"
      hidden={false}
      id="uuID3"
      iframeAttrs={Object {}}
      inline={false}
      mainAttrs={null}
      name={null}
      nestingLevel={null}
      noIndex={false}
      pureRender={false}
      ref_={null}
      resize={false}
      selected={false}
      src="https://unicorn.com"
      style={null}
      syncTimeout={1}
      tooltip={null}
    />
  </Column>
  <Column
    className={null}
    colWidth="s6"
    content={null}
    controlled={true}
    disabled={false}
    dynamic={false}
    footer={null}
    header={null}
    hidden={false}
    id="uuID4"
    ignoreInnerHTML={false}
    key="uuID4"
    level={null}
    mainAttrs={null}
    name={null}
    nestingLevel={null}
    noIndex={false}
    noSpacing={false}
    pureRender={false}
    ref_={null}
    selected={false}
    style={null}
    tooltip={null}
    underline={false}
    width={null}
  >
    <Iframe
      className={null}
      controlled={true}
      disabled={false}
      height="250"
      hidden={false}
      id="uuID5"
      iframeAttrs={Object {}}
      inline={false}
      mainAttrs={null}
      name={null}
      nestingLevel={null}
      noIndex={false}
      pureRender={false}
      ref_={null}
      resize={false}
      selected={false}
      src="https://plus4u.net"
      style={null}
      syncTimeout={1}
      tooltip={null}
    />
  </Column>
</div>
`;

exports[`UU5.Bricks.Iframe height 1`] = `
<div
  className="uu5-bricks-iframe"
  id="uuID"
  style={Object {}}
  title={null}
/>
`;

exports[`UU5.Bricks.Iframe height 2`] = `
<div
  className="uu5-bricks-iframe"
  id="uuID"
  style={Object {}}
  title={null}
/>
`;

exports[`UU5.Bricks.Iframe height 3`] = `
<div
  className="uu5-bricks-iframe"
  id="uuID"
  style={Object {}}
  title={null}
/>
`;

exports[`UU5.Bricks.Iframe height 4`] = `
<div
  className="uu5-bricks-iframe"
  id="uuID"
  style={Object {}}
  title={null}
/>
`;

exports[`UU5.Bricks.Iframe hidden 1`] = `
<div
  className="uu5-bricks-iframe uu5-common-hidden"
  id="uuID"
  style={Object {}}
  title={null}
/>
`;

exports[`UU5.Bricks.Iframe hidden 2`] = `
<div
  className="uu5-bricks-iframe"
  id="uuID"
  style={Object {}}
  title={null}
/>
`;

exports[`UU5.Bricks.Iframe hidden 3`] = `
<div
  className="uu5-bricks-iframe"
  id="uuID"
  style={Object {}}
  title={null}
/>
`;

exports[`UU5.Bricks.Iframe hidden 4`] = `
<div
  className="uu5-bricks-iframe uu5-common-hidden"
  id="uuID"
  style={Object {}}
  title={null}
/>
`;

exports[`UU5.Bricks.Iframe id 1`] = `
<div
  className="uu5-bricks-iframe"
  id="myId"
  style={Object {}}
  title={null}
/>
`;

exports[`UU5.Bricks.Iframe iframeAttrs 1`] = `
<div
  className="uu5-bricks-iframe"
  id="uuID"
  style={Object {}}
  title={null}
/>
`;

exports[`UU5.Bricks.Iframe inline 1`] = `
<div
  className="uu5-bricks-iframe uu5-bricks-iframe-inline"
  id="uuID"
  style={Object {}}
  title={null}
/>
`;

exports[`UU5.Bricks.Iframe inline 2`] = `
<div
  className="uu5-bricks-iframe"
  id="uuID"
  style={Object {}}
  title={null}
/>
`;

exports[`UU5.Bricks.Iframe inline 3`] = `
<div
  className="uu5-bricks-iframe"
  id="uuID"
  style={Object {}}
  title={null}
/>
`;

exports[`UU5.Bricks.Iframe inline 4`] = `
<div
  className="uu5-bricks-iframe uu5-bricks-iframe-inline"
  id="uuID"
  style={Object {}}
  title={null}
/>
`;

exports[`UU5.Bricks.Iframe mainAttrs 1`] = `
<div
  className="uu5-bricks-iframe"
  id="uuID"
  style={Object {}}
  title={null}
/>
`;

exports[`UU5.Bricks.Iframe name 1`] = `
<div
  className="uu5-bricks-iframe"
  id="uuID"
  name="myName"
  style={Object {}}
  title={null}
/>
`;

exports[`UU5.Bricks.Iframe nestingLevel 1`] = `
<div
  className="uu5-bricks-iframe"
  id="uuID"
  style={Object {}}
  title={null}
/>
`;

exports[`UU5.Bricks.Iframe nestingLevel 2`] = `
<div
  className="uu5-bricks-iframe"
  id="uuID"
  style={Object {}}
  title={null}
/>
`;

exports[`UU5.Bricks.Iframe nestingLevel 3`] = `
<div
  className="uu5-bricks-iframe"
  id="uuID"
  style={Object {}}
  title={null}
/>
`;

exports[`UU5.Bricks.Iframe nestingLevel 4`] = `
<div
  className="uu5-bricks-iframe"
  id="uuID"
  style={Object {}}
  title={null}
/>
`;

exports[`UU5.Bricks.Iframe nestingLevel 5`] = `
<div
  className="uu5-bricks-iframe"
  id="uuID"
  style={Object {}}
  title={null}
/>
`;

exports[`UU5.Bricks.Iframe nestingLevel 6`] = `
<div
  className="uu5-bricks-iframe"
  id="uuID"
  style={Object {}}
  title={null}
/>
`;

exports[`UU5.Bricks.Iframe nestingLevel 7`] = `
<div
  className="uu5-bricks-iframe"
  id="uuID"
  style={Object {}}
  title={null}
/>
`;

exports[`UU5.Bricks.Iframe nestingLevel 8`] = `
<div
  className="uu5-bricks-iframe"
  id="uuID"
  style={Object {}}
  title={null}
/>
`;

exports[`UU5.Bricks.Iframe nestingLevel 9`] = `
<div
  className="uu5-bricks-iframe"
  id="uuID"
  style={Object {}}
  title={null}
/>
`;

exports[`UU5.Bricks.Iframe nestingLevel 10`] = `
<div
  className="uu5-bricks-iframe"
  id="uuID"
  style={Object {}}
  title={null}
/>
`;

exports[`UU5.Bricks.Iframe nestingLevel 11`] = `
<div
  className="uu5-bricks-iframe"
  id="uuID"
  style={Object {}}
  title={null}
/>
`;

exports[`UU5.Bricks.Iframe nestingLevel 12`] = `
<div
  className="uu5-bricks-iframe"
  id="uuID"
  style={Object {}}
  title={null}
/>
`;

exports[`UU5.Bricks.Iframe nestingLevel 13`] = `
<div
  className="uu5-bricks-iframe"
  id="uuID"
  style={Object {}}
  title={null}
/>
`;

exports[`UU5.Bricks.Iframe nestingLevel 14`] = `
<div
  className="uu5-bricks-iframe"
  id="uuID"
  style={Object {}}
  title={null}
/>
`;

exports[`UU5.Bricks.Iframe nestingLevel 15`] = `""`;

exports[`UU5.Bricks.Iframe nestingLevel 16`] = `""`;

exports[`UU5.Bricks.Iframe nestingLevel 17`] = `""`;

exports[`UU5.Bricks.Iframe nestingLevel 18`] = `""`;

exports[`UU5.Bricks.Iframe nestingLevel 19`] = `""`;

exports[`UU5.Bricks.Iframe nestingLevel 20`] = `""`;

exports[`UU5.Bricks.Iframe noIndex 1`] = `
<div
  className="uu5-bricks-iframe uu5-noindex"
  id="uuID"
  style={Object {}}
  title={null}
/>
`;

exports[`UU5.Bricks.Iframe noIndex 2`] = `
<div
  className="uu5-bricks-iframe"
  id="uuID"
  style={Object {}}
  title={null}
/>
`;

exports[`UU5.Bricks.Iframe noIndex 3`] = `
<div
  className="uu5-bricks-iframe"
  id="uuID"
  style={Object {}}
  title={null}
/>
`;

exports[`UU5.Bricks.Iframe noIndex 4`] = `
<div
  className="uu5-bricks-iframe uu5-noindex"
  id="uuID"
  style={Object {}}
  title={null}
/>
`;

exports[`UU5.Bricks.Iframe pureRender 1`] = `
<div
  className="uu5-bricks-iframe"
  id="uuID"
  style={Object {}}
  title={null}
/>
`;

exports[`UU5.Bricks.Iframe pureRender 2`] = `
<div
  className="uu5-bricks-iframe"
  id="uuID"
  style={Object {}}
  title={null}
/>
`;

exports[`UU5.Bricks.Iframe pureRender 3`] = `
<div
  className="uu5-bricks-iframe"
  id="uuID"
  style={Object {}}
  title={null}
/>
`;

exports[`UU5.Bricks.Iframe pureRender 4`] = `
<div
  className="uu5-bricks-iframe"
  id="uuID"
  style={Object {}}
  title={null}
/>
`;

exports[`UU5.Bricks.Iframe resize 1`] = `
<div
  className="uu5-bricks-iframe"
  id="uuID"
  style={Object {}}
  title={null}
>
  <Loading
    className={null}
    colorSchema={null}
    content={null}
    controlled={true}
    disabled={false}
    dynamic={false}
    hidden={false}
    id={null}
    ignoreInnerHTML={false}
    inline={false}
    mainAttrs={null}
    name={null}
    nestingLevel={null}
    noIndex={false}
    pureRender={false}
    ref_={null}
    selected={false}
    style={null}
    tooltip={null}
  />
</div>
`;

exports[`UU5.Bricks.Iframe resize 2`] = `
<div
  className="uu5-bricks-iframe"
  id="uuID"
  style={Object {}}
  title={null}
>
  <Loading
    className={null}
    colorSchema={null}
    content={null}
    controlled={true}
    disabled={false}
    dynamic={false}
    hidden={false}
    id={null}
    ignoreInnerHTML={false}
    inline={false}
    mainAttrs={null}
    name={null}
    nestingLevel={null}
    noIndex={false}
    pureRender={false}
    ref_={null}
    selected={false}
    style={null}
    tooltip={null}
  />
</div>
`;

exports[`UU5.Bricks.Iframe resize 3`] = `
<div
  className="uu5-bricks-iframe"
  id="uuID"
  style={Object {}}
  title={null}
/>
`;

exports[`UU5.Bricks.Iframe resize 4`] = `
<div
  className="uu5-bricks-iframe"
  id="uuID"
  style={Object {}}
  title={null}
/>
`;

exports[`UU5.Bricks.Iframe selected 1`] = `
<div
  className="uu5-bricks-iframe uu5-common-selected"
  id="uuID"
  style={Object {}}
  title={null}
/>
`;

exports[`UU5.Bricks.Iframe selected 2`] = `
<div
  className="uu5-bricks-iframe"
  id="uuID"
  style={Object {}}
  title={null}
/>
`;

exports[`UU5.Bricks.Iframe selected 3`] = `
<div
  className="uu5-bricks-iframe"
  id="uuID"
  style={Object {}}
  title={null}
/>
`;

exports[`UU5.Bricks.Iframe selected 4`] = `
<div
  className="uu5-bricks-iframe uu5-common-selected"
  id="uuID"
  style={Object {}}
  title={null}
/>
`;

exports[`UU5.Bricks.Iframe src 1`] = `
<div
  className="uu5-bricks-iframe"
  id="uuID"
  style={Object {}}
  title={null}
/>
`;

exports[`UU5.Bricks.Iframe style 1`] = `
<div
  className="uu5-bricks-iframe"
  id="uuID"
  style={
    Object {
      "borderRadius": "2px",
      "padding": "8px",
    }
  }
  title={null}
/>
`;

exports[`UU5.Bricks.Iframe style 2`] = `
<div
  className="uu5-bricks-iframe"
  id="uuID"
  style={
    Object {
      "backgroundColor": "red",
      "color": "blue",
    }
  }
  title={null}
/>
`;

exports[`UU5.Bricks.Iframe style 3`] = `
<div
  className="uu5-bricks-iframe"
  id="uuID"
  style={
    Object {
      "backgroundColor": "red",
      "color": "blue",
    }
  }
  title={null}
/>
`;

exports[`UU5.Bricks.Iframe style 4`] = `
<div
  className="uu5-bricks-iframe"
  id="uuID"
  style={
    Object {
      "borderRadius": "2px",
      "padding": "8px",
    }
  }
  title={null}
/>
`;

exports[`UU5.Bricks.Iframe syncTimeout 1`] = `
<div
  className="uu5-bricks-iframe"
  id="uuID"
  style={Object {}}
  title={null}
/>
`;

exports[`UU5.Bricks.Iframe tooltip 1`] = `
<div
  className="uu5-bricks-iframe"
  id="uuID"
  style={Object {}}
  title="My Tooltip"
/>
`;
